pkg/patterns/addon/pkg/status/aggregate.go

```diff
@@ -46,50 +47,90 @@ type aggregator struct {
 func (a *aggregator) Reconciled(ctx context.Context, src declarative.DeclarativeObject, objs *manifest.Objects) error {
 	log := log.Log
 
-	instance, ok := src.(addonv1alpha1.CommonObject)
-	if !ok {
+	unstruct, ok := src.(*unstructured.Unstructured)
```

Comment:
It might be clearer if you constructed the `healthy` state first, and then applied it to the object.  i.e. move the type-checking on src to around line 89